src/service.coffee

```diff
@@ -145,14 +139,25 @@ module.exports = (settings)->
 
     es.random options
 
-  service.get '/:type/:id' , jsonP( (req)->
+  service.get '/:type/:id' , jsonP( (req, res, next) ->
     tf = settings.types[req.params.type]?.documentTransform
 
-    es.fetch(req.params.id,req.params.type).then (body)->
-      if tf? and tf.transform? then tf.transform body._source else body._source
+    es.fetch req.params.id, req.params.type
+      .then (body) ->
+        if tf? and tf.transform? then tf.transform body._source else body._source
+      .catch (error) ->
+        if settings.notFoundHandlers?
+          next()
+        else
+          console.error error
+          error
```

Comment:
If you `catch` the error, you should `throw` it instead of returning it. Otherweise IRMa will report status 200 which is clearly not what we want.